pandas/core/resample.py

```diff
@@ -2827,6 +2827,16 @@ def asfreq(
         if how is None:
             how = "E"
 
+        if isinstance(freq, str):
```

Comment:
a few lines down, there's ` obj.index.asfreq(freq, how=how)` - should these validation go in there?

Reply:
thanks, you are right. I moved the check `if isinstance(freq, str)` to the definition of `asfreq` in `pandas/core/indexes/period.py`